core/qmckcpp/src/qmck/extended_logic_table.cpp

```diff
@@ -4,10 +4,12 @@
 
 qmck::extended_logic_table::extended_logic_table(qmck::logic_table const &lhs)
 {
+    this->format = lhs.format;
+
     auto n = lhs.inputs.size();
     for (size_t i = 0; i < n; i++)
     {
-        if (__builtin_popcount(lhs.outputs[i])) // ignore zero output
+        if (lhs.outputs[i]) // ignore zero output
         {
             uint8_t rank = __builtin_popcount(lhs.inputs[i]);
```

Comment:
use the "auto + static_cast" idiom here to show that here is a wanted conversion. ( auto rank = static_cast<uint8_t>(__builtin_popcount(lhs.inputs[i])); )